include/range/v3/range_concepts.hpp

```diff
@@ -141,6 +142,25 @@ namespace ranges
                     ));
             };
 
+            struct ContiguousRange
+              : refines<RandomAccessRange>
+            {
+                template<typename Rng>
+                using data_reference_t = decltype(*ranges::data(std::declval<Rng&>()));
+
+                template<typename Rng>
+                using datum_t = meta::_t<std::remove_reference<data_reference_t<Rng>>>;
+
+                template<typename Rng>
+                auto requires_(Rng && rng) -> decltype(
+                    concepts::valid_expr(
+                        concepts::model_of<Same, InputRange::value_t<Rng>,
+                            meta::_t<std::remove_cv<datum_t<Rng>>>>(),
+                        concepts::model_of<CommonReference, InputRange::reference_t<Rng>,
+                            data_reference_t<Rng>>()
+                    ));
+            };
```

Comment:
Now that I think about it, I'm not sure `ContiguousRange` needs `ContiguousIterator` right now. Instead, once we get it we can add an overload of the `data` customization point that accepts ranges with iterators that satisfy `ContiguousIterator`. The `ContiguousRange` concept itself wouldn't need to change.